subprojects/docs/src/docs/userguide/dep-man/04-modeling-features/modeling_features.adoc

```diff
@@ -280,8 +280,103 @@ Contrary to <<#sec:maven-mapping-to-variants, Maven>>, there is no derivation st
 This means that when you have a dependency on an Ivy module, Gradle does _not_ perform variant aware resolution.
 Instead, it uses a legacy "compatibility" mode.
 
-[[adding-more-variants]]
-== Adding more variants to a component
+[[cross-project-publications]]
+== Sharing artifacts between projects
+
+A common pattern, in multi-project builds, is that one project consumes the artifacts of another project.
+In general, the simplest consumption form in the Java ecosystem is that when `A` depends on `B`, then `A` would depend on the `jar` produced by project `B`.
+We have explained in this chapter that in reality `A` depends on a _variant of_ `B` and that this variant depends on the needs of `A`.
+For compilation, we need the API dependencies of `B`, provided by the `apiElements` variant.
+For runtime, we need the runtime dependencies of `B`, provided by the `runtimeElements` variant.
+
+However, what if you need a _different_ artifact than the main one?
+Gradle provides, for example, built-in support for depending on the <<java_testing.adoc#sec:java_test_fixtures,test fixtures>> of another project, but sometimes the artifact you need to depend on simply isn't exposed as a variant.
+
+In order to be _safe to share_ between projects and allow maximum performance (parallelism), such artifacts must be exposed via _outgoing configurations_.
+
+[WARNING]
+.Don't reference other project tasks directly
+====
+It is frequent to see cross-project dependencies declared like this:
+
+```groovy
+dependencies {
+   // this is unsafe!
+   implementation project(":other").tasks.someOtherJar
+}
+```
+
+This publication model is _unsafe_ and can lead to non-reproducible and hard to parallelize builds.
+This section explains how to _properly create cross-project boundaries_ by defining "exchanges" between projects.
+Those "exchanges" are precisely _variants_.
+====
+
+There are two, complementary, options to share artifacts between projects.
+The <<#sec:simple-sharing-artifacts-between-projects,simplified version>> is only suitable if what you need to share is a simple artifact that doesn't depend on the consumer.
+It also is only limited to cases where this artifact shouldn't be published on a repository, and that the consumer itself wouldn't have a published dependency on this artifact.
```

Comment:
-> `The simple solution is also limited to cases where this artifact is not published to a repository. Which also implies that the consumer does not publish a dependency to this artifact.`